project/yxuan0329/proposal.rst

```diff
@@ -0,0 +1,171 @@
+distance data and arrange the results based on the temporal information may be
+useful for the users.
+
+The R-Tree Spatio-Temporal Search Engine aims to provide a search engine for 
+users to query multi-dimensional data with both spatial and temporal 
+information. In this project, we will provide a demonstration of the R-tree 
+search engine, enabling users to enter queries and receive a set of results 
+that include the nearest distance and the best time-matched entities.
+
+
+Problem to Solve
+================
+
+.. Describe the problem or the set of problems you want to solve.  Include
+.. necessary background information without making it lengthy.
+
+.. Some points may help you organize the problem description:
+
+.. 1. The field or industry of the problem.
+.. 2. The physics and/or the mathematics behind the problem.
+.. 3. The algorithm or numerical method that should be applied for solving the
+..    problem.
+
+R-tree is a data structure renowned for its exceptional performance in 
+searching multi-dimensional data, it plays a pivotal role in various 
+applications, particularly in spatial information systems such as maps. 
+However, beyond spatial information, data in map system should also take 
+temporal feature in considerations for better searching performance. When 
+searching nearest location on map system, users should be able to see different 
+sorting results depending on their specified searching time. In other words, 
+the users should be able to receive different searching results for the same 
+query at different days and times.
+
+The goal of this project is to integrate temporal data into the indexing 
+process, effectively creating a new temporal dimension within the R-tree 
+structure. We aspire to enable users to not only search by spatial attributes 
+but also to incorporate temporal considerations into their queries. 
+Furthermore, this project seeks to refine the organization of data within the 
+R-tree, prioritizing results based on temporal relevance, thereby delivering a 
+more comprehensive and responsive mapping experience.
+
+
+
+Prospective Users
+=================
+
+.. Describe the users of your software and how they will use it.  It is OK to
+.. combine this section with the previous one (`Problem to solve`_).
+
+This project aims to provide to users that want to search multidimensional data 
+including spatial and temporal features.
+
+System Architecture
+===================
+
+.. Analyze how your system takes input, produces results, provide interface, and
+.. performs any other operations.  Describe the system's work flow.  You may
+.. consider to use a flow chart but it is not required.  Specify the constraints
+.. assumed in your system.  Describe the modularization of the system.
+(Will be finished soon after discussion)
+
+
+Implementation Plan
+===================
+
+1. Frontend design:
+The search engine will provide a simple UI with one text block for users to 
```

Comment:
Making GUI (assuming it's what you meant by UI) for users is a lot of work.  It could take more time than what the project time frame may offer.